src/ol/renderer/webgl/shaders.js

```diff
@@ -0,0 +1,213 @@
+/**
+ * @module ol/renderer/webgl/shaders
+ */
+import {asArray} from '../../color.js';
+
+/**
+ * Attribute names used in the default shaders.
+ * @enum {string}
+ */
+export const DefaultAttributes = {
+  COLOR: 'color',
+  OPACITY: 'opacity',
+  WIDTH: 'width',
+};
+
+/**
+ * Packs red/green/blue channels of a color into a single float value; alpha is ignored.
+ * This is how DefaultAttributes.COLOR is expected to be computed.
+ * @param {import("../../color.js").Color|string} color Color as array of numbers or string
+ * @return {number} Float value containing the color
+ * @api
```

Comment:
Since this is experimental, I removed this `@api` annotation in an additional commit.